modules/wms/src/lib/data-sources/arcgis-image-service.ts

```diff
@@ -0,0 +1,90 @@
+// loaders.gl, MIT license
+
+import type {WMSCapabilities, WMSFeatureInfo, WMSLayerDescription} from '@loaders.gl/wms';
+import {ImageLoader, ImageType} from '@loaders.gl/images';
+import {LoaderOptions, path} from '@loaders.gl/loader-utils';
+
+type FetchLike = (url: string, options?: RequestInit) => Promise<Response>;
+
+export type ArcGISProps = {
+  url: string;
+  loadOptions?: LoaderOptions;
+  fetch?: typeof fetch | FetchLike;
+};
+
+export class ArcGISImageServerDataSource {
+  url: string;
+  loadOptions: LoaderOptions;
+  fetch: typeof fetch | FetchLike;
+
+  constructor(props: ArcGISProps) {
+    this.url = props.url;
+    this.loadOptions = props.loadOptions || {};
+    this.fetch = props.fetch || fetch;
+  }
+
+ // URL creators
+
+ metadataURL(options: {parameters?: Record<string, unknown>}): string {
+    return this.getUrl({path: '', ...options});
+  }
+
+  /** 
+   * Form a URL to an ESRI ImageServer
+   // https://sampleserver6.arcgisonline.com/arcgis/rest/services/NLCDLandCover2001/ImageServer/exportImage?bbox=${bounds[0]},${bounds[1]},${bounds[2]},${bounds[3]}&bboxSR=4326&size=${width},${height}&imageSR=102100&time=&format=jpgpng&pixelType=U8&noData=&noDataInterpretation=esriNoDataMatchAny&interpolation=+RSP_NearestNeighbor&compression=&compressionQuality=&bandIds=&mosaicRule=&renderingRule=&f=image`,
+   */
+  exportImageURL(options: {
```

Comment:
@jwasilgeo we should focus on this file. Adding stubs and parsers for ImageServer REST apis.